rules/prefer-query-selector.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+const getDocsUrl = require('./utils/get-docs-url');
+
+const forbiddenIdentifierNames = [
+	'getElementById',
+	'querySelectorAll',
+	'getElementsByClassName',
+	'getElementsByTagName'
+];
+
+const VALID_QUOTES = /([',",`])/;
+const getRange = (prop, node) => [prop.start, node.arguments[0].end];
+const getReplacement = (ctx, identifierName, node) => {
+	const argAsTxt = ctx.getSourceCode().getText(node.arguments[0]);
```

Comment:
It would be better to operate on the AST nodes instead of the text. Then you don't have to deal with possible quotes, escapes, etc, it gets complicated quickly. Looks at how some of the other rules are implemented, for inspiration.

Reply:
Yes i know that operating on text is not the best. I wanted to do this by returning an array of fixes one for changing `MemberExpression` and the other for changing arguments `Literal`. However, although the documentation says that it is possible to return an array of fixes i could not make it work.

Reply:
Can you push those changes somewhere? Hard to tell what's wrong otherwise.

It should look like something like this:

```js
context.report({
    node: node,
    message: "Missing semicolon",
    fix: function(fixer) {
        return [
        	fixer.insertTextAfter(node, ";"),
        	fixer.insertTextAfter(node, ";")
        ]
    }
});
```

It should be an array of only `fixing` objects, which are returned by the `fixer` methods, like `fixer.insertTextAfter`.

Reply:
Ok, @sindresorhus sorry if that takes too long but was investing this case with fixer returning an array. I believe it is because of our wrapper `eslint-ava-rule-tester` which has a dependency on old eslint package. I already made a PR and checked it locally. @jfmengels if you could review it here: https://github.com/jfmengels/eslint-ava-rule-tester/pull/11